app/main.py

```diff
@@ -263,62 +303,122 @@ def click_department_risks(feature, radio_button_value):
     to switch from the "Alerts and Infrastructure" to the "Risk Scores" view and vice-versa.
 
+    If a click is detected, it updates the "switch_map_view_1" object (an HTML placeholder built by via the Homepage
+    function which is defined in homepage.py) and this update will itself trigger the "change_map_style" callback
+    defined below.
+
+    NB: because two buttons (the one considered here and the alert banner in risks mode) can lead to a change of the map
+    view, the number of clicks on the map style button is not a reliable indicator of the map being viewed by the user.
+    """
+    if n_clicks is None:
+        raise PreventUpdate
+
+    return n_clicks
+
+
+@app.callback(
+    [Output('map_style_button', 'children'),
+     Output('hp_map', 'children'),
+     Output('hp_slider', 'children'),
+     Output('current_map_style', 'children')],
+    [Input('switch_map_view_1', 'children'),
+     Input('switch_map_view_2', 'children')],
+    [State('map_style_button', 'children'),
+     State('current_map_style', 'children')]
+)
+def change_map_style_main(map_style_button_input, alert_button_input, map_style_button_label, current_map_style):
+    """
+    -- Moving between alerts and risks views --
+
+    This callback is the one that actually updates the map view based on user's choice.
+
+    It can be triggered:
+    - either by a click on the main map style button in the user selection area on the left of the page;
+    - or by a click on the alert banner under risks mode.
+
+    This determines the two Inputs of this callback (see the two callbacks above).
+
+    This callback also takes as a State argument the map style currently being viewed.
+
     It relies on the choose_map_style function, imported from homepage.
 
     It returns:
 
     - the appropriate for the button on which the user has just clicked;
     - the right map object;
     - the slider object if relevant.
-    '''
-    if n_clicks is None:
-        n_clicks = 0
+    """
 
-    return choose_map_style(n_clicks)
+    # Deducing from the map style button label, the argument that we should pass to the choose_map_style function
+    if current_map_style == 'alerts':
+        arg = 1
 
+    elif current_map_style == 'risks':
+        arg = 0
 
-@app.callback(
-    [Output('map', 'center'),
-     Output('map', 'zoom')],
-    Input("alert_button", "n_clicks")
-)
-@cache.memoize()
-def change_zoom_center(n_clicks=None):
-    '''
-    -- Zooming on the alert from the banner --
+    ctx = dash.callback_context
 
-    This callback takes as input the number of clicks on the alert banner.
-    If the number of clicks is strictly above 0, it triggers a zoom on the alert marker.
-    It relies on the define_map_zoom_center function, imported from alerts.
-    '''
-    if n_clicks is None:
-        n_clicks = 0
+    if not ctx.triggered:
+        raise PreventUpdate
+
+    else:
+        input_id = ctx.triggered[0]['prop_id'].split('.')[0]
```

Comment:
Yes, exactly! I would add that it allows to figure out which input has triggered the callback _for the last time_ because I have the impression that the object stores all triggering events (at least for a given session I guess). But your understanding is accurate 😄